### DISCUSSION TITLE
Why moderator actions without the author?

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#5134] [Q&A] [answered]
Judging by [the commit](https://github.com/Chatterino/chatterino2/commit/e31dc09e91ea13dfec4fcc03946069f61b4b8b43), the nicknames should show, but I have all moderator actions shown without the author.

For example: "`nickname` has been timeout for 1m 50s", not "`nickname` has been timeout by `modname` for 1m 50s".

Why are there no moderator nicknames and can they be brought back?

**Comment by pajlada** [comment, marked as answer]
It's only supposed to work in channels where you are a broadcaster or moderator, otherwise we don't receive that information from Twitch.

I just confirmed it works for me as expected in a channel where I am a broadcaster.
![image](https://github.com/Chatterino/chatterino2/assets/962989/4367cee4-f75e-4ff2-a8b7-1deccd91ef3d)

**Reply by Mm2PL** [reply]
To expand on what pajlada said: Chatterino has two ways of acquiring ban or timeout information. We get this info from IRC which gives us a timeout length (or information that it was a ban), this is used for showing that a message was deleted. Then there is also PubSub which gives us more detailed moderator-only information like who applied the action and the reason they gave. IRC usually comes in first and PubSub message replaces that. That's why you might see a timeout pop up without the additional information and then have it added.
